src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -0,0 +1,427 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gcs_actor_scheduler.h"
+#include <ray/common/ray_config.h>
+#include <ray/protobuf/node_manager.pb.h>
+#include <ray/util/asio_util.h>
+#include "gcs_actor_manager.h"
+
+namespace ray {
+namespace gcs {
+
+class GcsWorkerLeaseImpl : public WorkerLeaseInterface {
```

Comment:
Hmm why implement this interface again instead of just using `RayletClient`?

Reply:
> Hmm why implement this interface again instead of just using `RayletClient`?

I will replace it with `RayletClient`, also, I will replace the `GcsCoreWorkerClientImpl` with `CoreWorkerClient`.

Reply:
Looks great!